### DISCUSSION TITLE
FastAPI TestClient error

### DISCUSSION THREAD

**Discussion by theinfosecguy** [fastapi/fastapi#9432] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi.TestClient import TestClient

@pytest.fixture
def setup_client():
    return TestClient(base_app, url)
```

### Description

I recently migrated my application to Python3.10 & I'm facing issues with FastAPI TestClient.

I'm getting following error when building the project:

`AttributeError: module ‘httpx’ has no attribute ‘BaseTransport’`

### Operating System

macOS

### Operating System Details

MacOS Ventura

### FastAPI Version

0.86.0

### Python Version

3.10

**Comment by mateoradman** [comment, marked as answer]
Hi @theinfosecguy,

There are multiple issues that I could identify.
1. your import is wrong. There is no fastapi.TestClient module. Please import the test client `from fastapi.testclient import TestClient`.
2. version 0.86.0 of FastAPI uses requests module for TestClient. Therefore, please make sure you have requests package installed `pip install requests`

Here is a minimal example that works. 

app/main.py
```python
from fastapi import FastAPI

app = FastAPI()


@app.get("/")
def read_root():
    return {"Hello": "World"}
```

app/test_main.py
```python
from fastapi.testclient import TestClient
import pytest
from .main import app


@pytest.fixture
def setup_client():
    return TestClient(app)


def test_root(setup_client):
    r = setup_client.get("/")
    assert r.ok
```
